code/modules/fabrication/_fabricator.dm

```diff
@@ -0,0 +1,279 @@
+/obj/machinery/fabricator
+	name = "autolathe"
+	desc = "A large device loaded with various item schematics. It produces common day to day items from a variety of materials."
+	icon = 'icons/obj/machinery/fabricators/autolathe.dmi'
+	icon_state = "autolathe"
+	density = TRUE
+	anchored = TRUE
+	use_power = POWER_USE_IDLE
+	idle_power_usage = 10
+	active_power_usage = 2000
+	clicksound = /singleton/sound_category/keyboard_sound
+	clickvol = 30
+	manufacturer = "hephaestus"
+
+	/// What location to print to. Only used for mounted autolathes
+	var/atom/print_loc
+
+	/// Class of fabricator. Determines recipes loaded. See entires in [__fabricator_defines.dm]
+	var/fabricator_class = FABRICATOR_CLASS_GENERAL
+	/// List of stored materials.
+	var/list/stored_material = list()
+	/// List of material capacities. Should not be modified directly, use [var/list/base_storage_capacity] instead.
+	var/list/storage_capacity = list()
+	/// Base storage capacity, which is modified by default by the amount and tier of matter bins.
+	var/list/base_storage_capacity = list(
+		DEFAULT_WALL_MATERIAL = 25000,
+		MATERIAL_ALUMINIUM = 25000,
+		MATERIAL_GLASS = 12500,
+		MATERIAL_PLASTIC = 12500
+	)
+	/// Current category to show for this fabricator
+	var/show_category = "All"
+
+	/// Status bitflags for the fabricator
+	var/fab_status_flags = 0
+
+	/// Current queue for this fabricator
+	var/list/print_queue = list()
+	/// What is currently printing in this fabricator
+	var/datum/fabricator_build_order/currently_printing
+
+	/// How efficient this fabricator uses materials. Modified by default by the amount and tier of manipulators
+	var/mat_efficiency = 1
+	/// What to multiply build times by. Modified by default by the amount and tier of manipulators
+	var/build_time_multiplier = 1
+
+	/// Snowflake for mounted autolathes
+	var/does_flick = TRUE
+
+	/// The fabricator's wires
+	var/datum/wires/fabricator/wires
+
+	component_types = list(
+		/obj/item/circuitboard/autolathe,
+		/obj/item/stock_parts/matter_bin = 3,
+		/obj/item/stock_parts/manipulator,
+		/obj/item/stock_parts/console_screen
+	)
+
+	/// List of stored reagents, for future usage (biogenerator, circuit printer)
+	var/static/list/stored_substances_to_names = list()
```

Comment:
Probably should be a GLOB, or not exist at all since I don't see it used anywhere besides being instantiated

Reply:
It's planned to be utilized for other fabricators, but I'll remove it until it's used.